includes/Classifai/Providers/OpenAI/Moderation.php

```diff
@@ -0,0 +1,337 @@
+<?php
+/**
+ * OpenAI Moderation integration
+ */
+
+namespace Classifai\Providers\OpenAI;
+
+use Classifai\Providers\Provider;
+
+use WP_Error;
+
+class Moderation extends Provider {
+
+	use OpenAI;
+
+	/**
+	 * OpenAI Moderation constructor.
+	 *
+	 * @param string $service The service this class belongs to.
+	 */
+	public function __construct( $service ) {
+		parent::__construct(
+			'OpenAI Moderation',
+			'Moderation',
+			'openai_moderation',
+			$service
+		);
+
+		// Set the onboarding options.
+		$this->onboarding_options = array(
+			'title'    => __( 'OpenAI Moderation', 'classifai' ),
+			'fields'   => array( 'api-key' ),
+			'features' => array(
+				'enable_moderation' => __( 'Enable comments moderation', 'classifai' ),
+			),
+		);
+	}
+
+	/**
+	 * Register what we need for the plugin.
+	 *
+	 * This only fires if can_register returns true.
+	 */
+	public function register() {
+		add_filter( 'comment_row_actions', [ $this, 'comment_row_actions' ], 10, 2 );
+		add_action( 'admin_init', [ $this, 'maybe_moderate_comment' ] );
+		add_filter( 'manage_edit-comments_columns', [ $this, 'add_comment_list_columns' ] );
+		add_action( 'manage_comments_custom_column', [ $this, 'add_comment_list_column_content' ], 10, 2 );
+		add_action( 'wp_insert_comment', [ $this, 'moderate_comment' ] );
+	}
+
+	/**
+	 * Prints custom column content
+	 *
+	 * @param string $column_name Column name
+	 * @param int    $comment_id  Column ID
+	 * @return void
+	 */
+	public function add_comment_list_column_content( $column_name, $comment_id ) {
+		if ( 'moderation_flagged' === $column_name ) {
+			$flagged = get_comment_meta( $comment_id, 'classifai_moderation_flagged', true );
+			if ( '0' === $flagged ) {
+				$flagged = 'No';
+			} elseif ( '1' === $flagged ) {
+				$flagged = 'Yes';
+			}
+			echo '<div style="text-align: center">' . esc_html( $flagged ) . '</div>';
+		}
+
+		if ( 'moderation_flags' === $column_name ) {
+			$flags = get_comment_meta( $comment_id, 'classifai_moderation_flags', true );
+			echo '<div style="text-align: center">' . esc_html( implode( ', ', $flags ) ) . '</div>';
+		}
+	}
+
+	/**
+	 * Prints custom column header
+	 *
+	 * @param array $columns Columns
+	 * @return array
+	 */
+	public function add_comment_list_columns( $columns ) {
+		$columns['moderation_flagged'] = __( 'Moderation Flagged' );
+		$columns['moderation_flags']   = __( 'Moderation Flags' );
+
+		return $columns;
+	}
+
+	/**
+	 * Moderates comment when clicked on "Moderate" action button in comment list
+	 *
+	 * @return void
+	 */
+	public function maybe_moderate_comment() {
+		$action     = sanitize_text_field( $_GET['a'] ?? null );
+		$comment_id = sanitize_text_field( $_GET['c'] ?? null );
+		$nonce      = sanitize_text_field( $_GET['nonce'] ?? null );
+
+		if (
+			'moderate' === $action &&
+			$comment_id &&
+			wp_verify_nonce( $nonce, 'moderate_comment' )
+		) {
+			$this->moderate_comment( $comment_id );
+			wp_redirect( '/wp-admin/edit-comments.php' );
```

Comment:
Since we're redirecting on the same domain, it would be good to use `wp_safe_redirect()`